### ISSUE TITLE
Feature Request: Add submission-song to song-search indexing config 

### ISSUE THREAD

**Issue by rosibaj** [icgc-argo/workflow-roadmap#49]
We have set up Submission Songs/Score specifically for data submitters to submit raw data. 

## Detailed Description
URLS 
```
Song DEV/QA
submission-song.rdpc-dev.cancercollaboratory.org 
submission-song.rdpc-qa.cancercollaboratory.org 
Score DEV/QA
submission-score.rdpc-dev.cancercollaboratory.org 
submission-score.rdpc-qa.cancercollaboratory.org 
Song PROD
submission-song.rdpc.cancercollaboratory.org 
Score PROD
submission-score.rdpc.cancercollaboratory.org 
```
## Expected Outcome 
RDPC gatway-api should be able to query for `sequencing_experiment` analyses that will be submitted to submission-song by data submitters.

**Comment by rosibaj** [comment]
Blocked by network policy {?)

**Comment by andricDu** [comment]
It looks like this is ready to be tested.

**Comment by rosibaj** [comment]
I tested this on RDPC environment 

- i successfulyl retireved a list of all studies from QA submission song
- I successfully retrieved a list of all analysis in the study DASH-CA (both published and unpublised) 
- when i tried to publishand analysis i gor a 500-internal server error 
- to make sure i was publishing an analysis that should be able to be published, i unpublished a published analysis `a71b9099-ed67-4c86-9b90-99ed67ec860a`.  this worked fine. 
- i attempted to republished this analysis and got another 500 error: 
```
curl --location --request PUT 'https://submission-song.rdpc-qa.cancercollaboratory.org/studies/DASH-CA/analysis/publish/a71b9099-ed67-4c86-9b90-99ed67ec860a' \
--header 'Authorization: Bearer ---------'
```

```
{
  "errorId" : "unknown.error",
  "httpStatusCode" : 500,
  "httpStatusName" : "INTERNAL_SERVER_ERROR",
  "requestUrl" : "http://submission-song.rdpc-qa.cancercollaboratory.org/studies/DASH-CA/analysis/publish/a71b9099-ed67-4c86-9b90-99ed67ec860a",
  "datetime" : "2020-11-20T00:41:32.78Z[GMT]",
  "timestamp" : 1605832892780,
  "message" : "Request processing failed; nested exception is org.springframework.web.client.HttpClientErrorException$Forbidden: 403 null",
  "debugMessage" : "[ROOT_CAUSE] -> org.springframework.web.client.HttpClientErrorException$Forbidden: 403 null",
  "stackTrace" : [
        "org.springframework.web.util.NestedServletException: Request processing failed; nested exception is org.springframework.web.client.HttpClientErrorException$Forbidden: 403 null",
        "at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1013)",
        "at org.springframework.web.servlet.FrameworkServlet.doPut(FrameworkServlet.java:919)",
        "at javax.servlet.http.HttpServlet.service(HttpServlet.java:663)",
        "at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:882)",
        "at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)",
        "at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)",
        "at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)",
        "at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)",
        "at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)",
        "at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)",
        "at org.springframework.boot.actuate.web.trace.servlet.HttpTraceFilter.doFilterInternal(HttpTraceFilter.java:88)",
        "at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:118)",
        "at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)",
        "at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)",
        "at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:320)",
        "at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:127)",
        "at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:91)",
        "at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)",
        "at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:119)",
        "at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)",
        "at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:137)",
        "at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)",
        "at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:111)",
        "at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)",
        "at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:170)",
        "at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)",
        "at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63)",
        "at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)",
        "at org.springframework.security.oauth2.provider.authentication.OAuth2AuthenticationProcessingFilter.doFilter(OAuth2AuthenticationProcessingFilter.java:176)",
        "at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)",
        "at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:116)",
        "at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)",
        "at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:74)",
        "at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:118)",
        "at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)",
        "at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:105)",
        "at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)",
        "at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:56)",
        "at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:118)",
        "at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)",
        "at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:215)",
        "at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:178)",
        "at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:357)",
        "at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:270)",
        "at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)",
        "at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)",
        "at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:99)",
        "at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:118)",
        "at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)",
        "at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)",
        "at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:92)",
        "at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:118)",
        "at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)",
        "at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)",
        "at org.springframework.web.filter.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:93)",
        "at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:118)",
        "at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)",
        "at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)",
        "at org.springframework.boot.actuate.metrics.web.servlet.WebMvcMetricsFilter.filterAndRecordMetrics(WebMvcMetricsFilter.java:114)",
        "at org.springframework.boot.actuate.metrics.web.servlet.WebMvcMetricsFilter.doFilterInternal(WebMvcMetricsFilter.java:104)",
        "at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:118)",
        "at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)",
        "at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)",
        "at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:200)",
        "at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:118)",
        "at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)",
        "at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)",
        "at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)",
        "at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)",
        "at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:490)",
        "at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)",
        "at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)",
        "at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)",
        "at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)",
        "at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)",
        "at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:408)",
        "at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)",
        "at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:853)",
        "at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1587)",
        "at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)",
        "at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)",
        "at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)",
        "at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)",
        "at java.base/java.lang.Thread.run(Unknown Source)",
        "Caused by: org.springframework.web.client.HttpClientErrorException$Forbidden: 403 null",
        "at org.springframework.web.client.HttpClientErrorException.create(HttpClientErrorException.java:83)",
        "at org.springframework.web.client.DefaultResponseErrorHandler.handleError(DefaultResponseErrorHandler.java:122)",
        "at org.springframework.web.client.DefaultResponseErrorHandler.handleError(DefaultResponseErrorHandler.java:102)",
        "at org.springframework.web.client.ResponseErrorHandler.handleError(ResponseErrorHandler.java:63)",
        "at org.springframework.web.client.RestTemplate.handleResponse(RestTemplate.java:778)",
        "at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:736)",
        "at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:710)",
        "at org.springframework.web.client.RestTemplate.exchange(RestTemplate.java:598)",
        "at bio.overture.song.server.service.StorageService.lambda$doGetString$0(StorageService.java:115)",
        "at org.springframework.retry.support.RetryTemplate.doExecute(RetryTemplate.java:287)",
        "at org.springframework.retry.support.RetryTemplate.execute(RetryTemplate.java:164)",
        "at bio.overture.song.server.service.StorageService.doGetString(StorageService.java:110)",
        "at bio.overture.song.server.service.StorageService.doGetBoolean(StorageService.java:121)",
        "at bio.overture.song.server.service.StorageService.isObjectExist(StorageService.java:68)",
        "at bio.overture.song.server.service.analysis.AnalysisServiceImpl.confirmUploaded(AnalysisServiceImpl.java:442)",
        "at bio.overture.song.server.service.analysis.AnalysisServiceImpl.lambda$checkMissingFiles$5(AnalysisServiceImpl.java:404)",
        "at java.base/java.util.stream.ReferencePipeline$2$1.accept(Unknown Source)",
        "at java.base/java.util.Collections$2.tryAdvance(Unknown Source)",
        "at java.base/java.util.Collections$2.forEachRemaining(Unknown Source)",
        "at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)",
        "at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)",
        "at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source)",
        "at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)",
        "at java.base/java.util.stream.ReferencePipeline.collect(Unknown Source)",
        "at bio.overture.song.server.service.analysis.AnalysisServiceImpl.checkMissingFiles(AnalysisServiceImpl.java:404)",
        "at bio.overture.song.server.service.analysis.AnalysisServiceImpl.publish(AnalysisServiceImpl.java:313)",
        "at bio.overture.song.server.service.analysis.AnalysisServiceImpl$$FastClassBySpringCGLIB$$a1a34d52.invoke(<generated>)",
        "at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)",
        "at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:749)",
        "at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)",
        "at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:295)",
        "at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:98)",
        "at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)",
        "at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:688)",
        "at bio.overture.song.server.service.analysis.AnalysisServiceImpl$$EnhancerBySpringCGLIB$$3b74ddbb.publish(<generated>)",
        "at bio.overture.song.server.service.analysis.AnalysisServiceSender.publish(AnalysisServiceSender.java:56)",
        "at bio.overture.song.server.controller.AnalysisController.publishAnalysis(AnalysisController.java:121)",
        "at bio.overture.song.server.controller.AnalysisController$$FastClassBySpringCGLIB$$b8162298.invoke(<generated>)",
        "at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)",
        "at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:749)",
        "at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)",
        "at org.springframework.security.access.intercept.aopalliance.MethodSecurityInterceptor.invoke(MethodSecurityInterceptor.java:69)",
        "at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)",
        "at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:688)",
        "at bio.overture.song.server.controller.AnalysisController$$EnhancerBySpringCGLIB$$1dc58364.publishAnalysis(<generated>)",
        "at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)",
        "at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)",
        "at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)",
        "at java.base/java.lang.reflect.Method.invoke(Unknown Source)",
        "at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:190)",
        "at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:138)",
        "at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:104)",
        "at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:892)",
        "at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:797)",
        "at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)",
        "at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1039)",
        "at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:942)",
        "at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1005)",
        "... 83 more",
        ""
    ]
}

```

**Comment by andricDu** [comment]
@rosibaj this was due to expired tokens. They been refreshed. You can try again.

**Comment by rosibaj** [comment]
Tested Nove 24, morning in RDOC QA environment.  Same issue. Still blocked.

**Comment by rosibaj** [comment]
Tested Nov 24th, at 1:00. 
- uploaded new analysis to ARGO QA Submission Song under the program DASH-CA 
- I can see the `newly submitted` analysis being indexed. 
- however, it looks like the information is not 100% accurate.  The rdpc api says the payload is `UNPUBLISHED` when it is in fact published: 

![image](https://user-images.githubusercontent.com/15824252/100135658-f4dba580-2e57-11eb-8a5a-145230d52276.png)
![image](https://user-images.githubusercontent.com/15824252/100135715-091fa280-2e58-11eb-908f-4226d78a6022.png)

Discussed with @andricDu  on slack.  Since the inital index and the index of analysis when submitted is working, we will consider this ticket as closed.  a followup ticket has been made to investigate the Song configurations and the publish issue noted above.

**Comment by rosibaj** [comment]
Followup issue: https://github.com/icgc-argo/workflow-roadmap/issues/73
